core/domain/exp_jobs_one_off_test.py

```diff
@@ -668,6 +669,38 @@ def test_migration_job_skips_deleted_explorations(self):
         with self.assertRaisesRegexp(Exception, 'Entity .* not found'):
             exp_services.get_exploration_by_id(self.NEW_EXP_ID)
 
+    def test_migration_job_creates_appropriate_classifier_models(self):
+        """Tests that the exploration migration job skips deleted explorations
+        and does not attempt to migrate.
+        """
+        self.save_new_exp_with_states_schema_v0(
+            self.NEW_EXP_ID, self.albert_id, self.EXP_TITLE)
+        exploration = exp_services.get_exploration_by_id(self.NEW_EXP_ID)
+
+        # Store classifier model for the new exploration.
+        classifier_model_id = classifier_models.ClassifierTrainingJobModel.create(
+            'TextClassifier', 'TextInput', self.NEW_EXP_ID, exploration.version,
+            datetime.datetime.utcnow(), {}, exploration.states.keys()[0],
+            feconf.TRAINING_JOB_STATUS_COMPLETE, None, 1)
+        # Store training job model for the classifier model.
+        classifier_models.TrainingJobExplorationMappingModel.create(
+            self.NEW_EXP_ID, exploration.version, exploration.states.keys()[0],
+            classifier_model_id)
+
+        # Start migration job on sample exploration.
+        job_id = exp_jobs_one_off.ExplorationMigrationJobManager.create_new()
+        exp_jobs_one_off.ExplorationMigrationJobManager.enqueue(job_id)
+        with self.swap(feconf, 'MIN_TOTAL_TRAINING_EXAMPLES', 0):
```

Comment:
Hmm, I thought about that but in schema 0 we didn't really have a notion of state or training data. To be frank, I am unable to figure out what goes where when that schema is updated to the latest version and hence I refrained from putting training data there. Maybe I will have to create a similar function which creates an exploration of a more recent schema version, where we do have a notion of training data. Also, training data is not part of such default explorations, so what it needs is a highly specific function (that creates an exploration with training data and also creates appropriate classifier data and mapping models) which is useful only in this particular test case.

Reply:
I don't think you need to start at schema 0. You can start at the schema where the ML fields are introduced (as I think you're suggesting).

Also, I think creating such setup functions in test_utils.py would probably be useful, since we're likely to have more backend tests for the ML stuff down the road. So it probably will have usefulness beyond this particular case.

Reply:
Modified test case.